diff -r 8e6e810e89ce toke.c
--- a/toke.c	Wed Mar 23 16:04:17 2011 -0400
+++ b/toke.c	Sun Mar 27 09:16:26 2011 -0400
@@ -3306,6 +3306,7 @@
 	}
 	if (stat(path, &sb) != 0 || !S_ISREG(sb.st_mode)) {
 	    efree(path);
+	    path = NULL;
 	    continue;
 	}
 	pl = malloc(sizeof(*pl));
diff -r 8e6e810e89ce toke.l
--- a/toke.l	Wed Mar 23 16:04:17 2011 -0400
+++ b/toke.l	Sun Mar 27 09:16:26 2011 -0400
@@ -639,6 +639,7 @@
 	}
 	if (stat(path, &sb) != 0 || !S_ISREG(sb.st_mode)) {
 	    efree(path);
+	    path = NULL;
 	    continue;
 	}
 	pl = malloc(sizeof(*pl));
