diff -urNp -x '*.orig' sudo-1.9.6p1.org/plugins/sudoers/sudoers.in sudo-1.9.6p1/plugins/sudoers/sudoers.in
--- sudo-1.9.6p1.org/plugins/sudoers/sudoers.in	2021-03-13 16:46:14.000000000 +0100
+++ sudo-1.9.6p1/plugins/sudoers/sudoers.in	2021-03-16 14:07:11.051018031 +0100
@@ -62,6 +62,9 @@
 ## Uncomment to send mail if the user does not enter the correct password.
 # Defaults mail_badpass
 ##
+# Uncomment to disable TTY tickets
+# Defaults !tty_tickets
+#
 ## Uncomment to enable logging of a command's output, except for
 ## sudoreplay and reboot.  Use sudoreplay to play back logged sessions.
 # Defaults log_output
