From https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=762465

Index: sudo-1.8.13/plugins/sudoers/boottime.c
===================================================================
--- sudo-1.8.13.orig/plugins/sudoers/boottime.c	2015-03-18 18:05:51.000000000 +0100
+++ sudo-1.8.13/plugins/sudoers/boottime.c	2015-05-14 14:48:33.855294076 +0200
@@ -79,6 +79,8 @@ get_boottime(struct timespec *ts)
     if (fp != NULL) {
 	while ((len = getline(&line, &linesize, fp)) != -1) {
 	    if (strncmp(line, "btime ", 6) == 0) {
+		if (line[len - 1] == '\n')
+		    line[len - 1] = '\0';
 		long long llval = strtonum(line + 6, 1, LLONG_MAX, NULL);
 		if (llval > 0) {
 		    ts->tv_sec = (time_t)llval;
