--- sudo-1.8.23/configure.ac.orig	2018-04-29 21:59:31.000000000 +0200
+++ sudo-1.8.23/configure.ac	2018-07-23 22:20:50.801696342 +0200
@@ -1552,7 +1552,7 @@
     *)		devsearch="$enableval"
 		;;
 esac])
-ds="`echo \"$devsearch\"|sed 's@/dev/*\([[^:]]*:*\)@_PATH_DEV \"\1\" @g'`"
+ds="$(echo "$devsearch"|sed 's@/dev/*\([[^:]]*:*\)@_PATH_DEV \"\1\" @g')"
 SUDO_DEFINE_UNQUOTED(_PATH_SUDO_DEVSEARCH, $ds)
 
 AC_ARG_WITH(selinux, [AS_HELP_STRING([--with-selinux], [enable SELinux support])],
