diff -dur -x '*~' -x '*.orig' -x '*.rej' sudo-1.8.13.orig/plugins/sudoers/env.c sudo-1.8.13/plugins/sudoers/env.c
--- sudo-1.8.13.orig/plugins/sudoers/env.c	2015-03-18 18:05:51.000000000 +0100
+++ sudo-1.8.13/plugins/sudoers/env.c	2015-03-25 09:47:10.000000000 +0100
@@ -197,12 +197,15 @@
 static const char *initial_keepenv_table[] = {
     "COLORS",
     "DISPLAY",
+    "HOME",
     "HOSTNAME",
     "KRB5CCNAME",
     "LS_COLORS",
+    "MAIL",
     "PATH",
     "PS1",
     "PS2",
+    "SSH_AUTH_SOCK",
     "XAUTHORITY",
     "XAUTHORIZATION",
     "XDG_CURRENT_DESKTOP",
