From dfee181d1579a24c679eff960c85bc659c56993f Mon Sep 17 00:00:00 2001
From: "Todd C. Miller" <Todd.Miller@sudo.ws>
Date: Sat, 11 Jun 2022 16:59:18 -0600
Subject: [PATCH] Fix compilation on Linux/x32; GitHub issue #158

---
 src/exec_ptrace.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/exec_ptrace.h b/src/exec_ptrace.h
index 4a5a86581..4340a48ea 100644
--- a/src/exec_ptrace.h
+++ b/src/exec_ptrace.h
@@ -64,8 +64,10 @@
  */
 #if defined(__x86_64__)
 # define SECCOMP_AUDIT_ARCH	AUDIT_ARCH_X86_64
-# define X32_execve		__X32_SYSCALL_BIT + 520
-# define X32_execveat		__X32_SYSCALL_BIT + 545
+# ifndef __ILP32__
+#  define X32_execve		__X32_SYSCALL_BIT + 520
+#  define X32_execveat		__X32_SYSCALL_BIT + 545
+# endif
 # define sudo_pt_regs		struct user_regs_struct
 # define reg_syscall(x)		(x).orig_rax
 # define reg_retval(x)		(x).rax
